game-core/src/test/java/games/strategy/triplea/delegate/battle/casualty/CasualtyOrderOfLossesTestOnBigWorldV3.java

```diff
@@ -165,4 +161,115 @@ void infantryAndArtillery() {
     assertThat(
         "Artillery has the better total power", result.get(1).getType(), is(testData.artillery));
   }
+
+  @Test
+  void nonAmphibiousMarineWithAmphibiousAssault() {
+    testData.addTech(new ImprovedArtillerySupportAdvance(testData.gameData));
+
+    final List<Unit> amphibUnits = new ArrayList<>();
+    amphibUnits.addAll(testData.tank(1));
+    amphibUnits.addAll(testData.artillery(1));
+    amphibUnits.addAll(testData.marine(1));
+
+    amphibUnits.forEach(
+        unit -> {
+          unit.getProperty(Unit.UNLOADED_AMPHIBIOUS)
+              .ifPresent(
+                  property -> {
+                    try {
+                      property.setValue(true);
+                    } catch (final MutableProperty.InvalidValueException e) {
+                      // should not happen
+                    }
+                  });
+        });
+
+    final List<Unit> attackingUnits = new ArrayList<>(amphibUnits);
+    attackingUnits.addAll(testData.marine(1));
+
+    final List<Unit> result =
+        CasualtyOrderOfLosses.sortUnitsForCasualtiesWithSupport(
+            CasualtyOrderOfLosses.Parameters.builder()
+                .targetsToPickFrom(attackingUnits)
+                .combatModifiers(
+                    CombatModifiers.builder().defending(false).territoryEffects(List.of()).build())
+                .player(testData.british)
+                .enemyUnits(List.of())
+                .battlesite(testData.france)
+                .costs(testData.costMap)
+                .data(testData.gameData)
+                .build());
+
+    assertThat(result, hasSize(4));
+    assertThat(
+        "Non amphibious marine only has attack of 2 since it doesn't get marine bonus",
+        result.get(0),
+        is(attackingUnits.get(3)));
+    assertThat(result.get(1), is(attackingUnits.get(1)));
+    assertThat("Amphibious marine has attack of 3", result.get(2), is(attackingUnits.get(2)));
+    assertThat(result.get(3), is(attackingUnits.get(0)));
```

Comment:
I don't know why the armor is the last unit.  This test has the same units as the `improvedArtillery` test except that one of the marines isn't amphibious.  I expected the tank to be below the amphibious marine like in the `improvedArtillery` but for some reason it is above.  I'm not entirely sure if this is correct behavior or some bug.

Reply:
## 1st Casualty

The first casualty is clear to remove the non-amphib.

## 2nd Casualty

This leaves the follow powers (where the +1 is support bonus):

```
tank      3
marine    3 +1
artillery 2
```

f the artillery is taken, leaves total of 6
```
tank 3
marine 3
```

if the marine is taken, leaves total of 5
```
tank 3
artillery 2
```

if the tank is taken, leaves total of 6
```
marine 3+1
artillery 2
```

Removing the marine is a bad choice. This leaves a choice between artillery and tank with the same remaining power. Tank is more expensive and has better defensive power, so the choice is to remove artillery.

## 3rd Casualty

We are left with this choice:
```
tank 3
marine 3
```
Tank is more expensive and has better defensive value, so marine is the choice

## 4th casualty

Trivial case as we are left with only the tank.


## Final ordering

marine, artillery, marine, tank